src/main/java/org/jenkinsci/plugins/pipeline/utility/steps/conf/WriteYamlStep.java

```diff
@@ -55,12 +55,10 @@
     private Object data;
     private String charset;
     private boolean overwrite;
+    private boolean returnText;
 
     @DataBoundConstructor
-    public WriteYamlStep(@Nonnull String file, @Nonnull Object data) {
-        if (isBlank(file)) {
-            throw new IllegalArgumentException("file parameter must be provided to writeYaml");
-        }
+    public WriteYamlStep(String file, @Nonnull Object data) {
```

Comment:
Just removing the `@Nonnull` won't help. Since the parameter is in the databound constructor it is still interpreted as mandatory.
Instead remove it from the databound constructor and add a databound setter for the field.
Keep the original constructor signature but deprecate it.

Reply:
Just as was done in #87

Reply:
Done.